### DISCUSSION TITLE
Building SMPlayer for macOS?

### DISCUSSION THREAD

**Discussion by cristim** [smplayer-dev/smplayer#91] [General]
Hello,

I wonder what would it take to get SMPlayer built on macOS, has anyone tried this?

Thanks!

**Comment by smplayer-dev** [comment]
We would need someone with a macOS computer, which can compile SMPlayer, fix its problems, maintain it...
I can compile it and create a dmg file here at github, but then I have no way to test it.

You can test yourself the file I've just built:
https://github.com/smplayer-dev/smplayer/actions/runs/838116234

You also need to install mplayer/mpv on your own.

**Reply by cristim** [reply]
Thanks, I have an M1 Macbook with mpv installed. I'll try this out over the weekend and also try to make a brew cask or formula for it.

**Comment by smplayer-dev** [comment]
I installed mac os in a virtual machine. The main problem is that both mplayer and mpv are displayed in a different window. I backported some code from umplayer (an old smplayer fork) which had this problem fixed for mplayer, but the code it's for Qt 4 and it crashes in Qt 5, with this message: `Cannot make QOpenGLContext current in a different thread`. I'm afraid I don't know how to fix it.

That problem could probably be fixed for mpv, by using libmpv but that requires major changes in smplayer.

My latest build for mac os: https://github.com/smplayer-dev/smplayer/actions/runs/856563978

**Reply by cristim** [reply]
thanks! I just looked at your initial build and it indeed failed to run mpv. In the command it used I noticed some no longer supported command-line flags related to software volume control that made it fail, so I had to disable that in the preferences.

I'll try the second build shortly.

**Reply by cristim** [reply]
Just tried the second build and I get a similar failure to run mpv but no also crash. If I take the mpv command and clean it up a bit it opens the video but mpv fails to play my video.

**Reply by smplayer-dev** [reply]
What version of mpv are you using?

**Reply by cristim** [reply]
I think the latest available currently in brew:
```
mpv 0.33.1 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on Mon Apr  5 10:16:25 PDT 2021
FFmpeg library versions:
   libavutil       56.51.100 (runtime 56.70.100)
   libavcodec      58.91.100 (runtime 58.134.100)
   libavformat     58.45.100 (runtime 58.76.100)
   libswscale      5.7.100 (runtime 5.9.100)
   libavfilter     7.85.100 (runtime 7.110.100)
   libswresample   3.7.100 (runtime 3.9.100)
FFmpeg version: 4.4
`
``

**Reply by smplayer-dev** [reply]
What options did you have to remove?

**Reply by cristim** [reply]
I deleted those regarding `softvol` by disabling it in the settings, and then manually deleted the `--term-playing-msg` which required a lot of inputs coming from smplayer which I didn't have in the CLI.

**Reply by smplayer-dev** [reply]
What error do you get with softvol?

**Reply by cristim** [reply]
```
Error parsing option softvol (option not found)
Setting commandline option --softvol=yes failed.
```

I also get these warnings after I remove it:
```
Warning: option --ass-line-spacing was replaced with --sub-ass-line-spacing and might be removed in the future.
Warning: option --sub-text-shadow-color was replaced with --sub-shadow-color and might be removed in the future.
Warning: option --softvol-max was replaced with --volume-max and might be removed in the future.
```

**Reply by smplayer-dev** [reply]
Ok, I'll try to fix those problems.

Does corevideo in mplayer work for you? I only get a green screen.

**Reply by cristim** [reply]
I'm using it with mpv, and it works fine.

I do have some video files that have playback issues but it also happens in QuickTime and VLC, so I guess it must be something else.

**Reply by smplayer-dev** [reply]
The problem with softvol shouldn't happen. It's only used when the option `--volume-max` is not available (older versions of mpv).

**Reply by cristim** [reply]
As i said I'm running 0.33.1, which is the latest version currently available at https://github.com/mpv-player/mpv/releases 

Looks like `--softvol` was deprecated in 0.29.0 which was released on 22 Jul 2018

```
options: drop deprecated --videotoolbox-format, --ff-aid, --ff-vid, --ff-sid, --ad-spdif-dtshd, --softvol options
```

**Reply by smplayer-dev** [reply]
The problem with `--ass-line-spacing` and `--sub-text-shadow-color` was already fixed some time ago. It seems smplayer didn't recognize your version of mpv correctly. Try to remove the file `player_info.ini` in the smplayer configuration folder, that'll make smplayer to try to identify mpv again.

**Comment by smplayer-dev** [comment]
The version in [this page](https://github.com/smplayer-dev/smplayer-build/actions/runs/954096211) (the artifact at the bottom) can now display the video image in the smplayer window when using "corevideo" or "shm" as video output. 

Known problems:
* some times the video image doesn't appear until you click on the video area or resize the window
* some times the floating control in fullscreen doesn't go away
* the computer power saving is not turned off
* support for mpv has been disabled

If you find other issues, let me know and I'll try to fix them.

**Comment by smplayer-dev** [comment]
[Version 21.1.0.9765](https://github.com/smplayer-dev/smplayer/releases/tag/v21.1.0.9765)
* Added support for mpv

The video is displayed in the smplayer window when using "Shared memory", "corevideo" or "sharedbuffer" as video output.
This is almost finished.

**Comment by easior** [comment]
I found the released SMPlayer for macOS was required 10.14 at least. Could anyone build SMPlayer compatible to macOS 10.13?

**Comment by smplayer-dev** [comment]
Actually I think SMPlayer can run on macOS 10.13, the problem is that mpv and mplayer included in the package don't work on macOS 10.13 so you actually need to build mpv or mplayer.
